api/catalogue/schema.graphql

```diff
@@ -0,0 +1,547 @@
+  It's a bit more to type, but the `and`-operation is more explicit.
+  """
+  and: [Filter!]
+
+  """
+  ### Boolean 'OR' filter
+
+  This recursive filter will `or` together all expressions given to it
+
+  ### Examples
+
+  Return all manifests where `uploadFilename` contains `some-file.sgy` _or_
+  `uploadFilename` contains `some-file-2`, by combining two `eq` filters:
+  ```
+  manifests(where: {
+    or: [
+      { eq: { uploadFilename: "some-file.sgy" } },
+      { eq: { uploadFilename: "some-file-2.sgy" } }
+    ]
+  } ) {
+    guid
+  }
+  ```
+
+  By utilizing the recursive property of the `or` filter, we can define the
+  following query, which return manifests where `uploadFilename` contains
+  `some-file.sgy` and `formatVersion` contains `1`, or manifests where
+  `uploadFilename` contains `some-file-2.sgy`
+  ```
+  manifests(where: {
+    or: [
+      { and: [
+        { eq: { uploadFilename: "some-file.sgy" } },
+        { eq: { formatVersion: 1 } }
+      ] },
+      { eq: { uploadFilename: "some-file-2.sgy" } },
+    ]
+  } ) {
+    guid
+  }
+  ```
+  """
+  or:  [Filter!]
+  #TODO implement not, lt, le, gt & ge
+}
+
+"""
+The input-version of the `Manifest`-type
+"""
+
+input ManfestInput {
+  formatVersion: Int
+  uploadFilename: String
+  guid: String
+  data: [DataInput!]
+  attributes: [AttributeInput!]
+  lineNumbers: [[Int!]!]
+  lineLabels: [String!]
+  sampleValueMin: Float
+  sampleValueMax: Float
+}
+
+"""
+The input-version of the `Data`-type
+"""
+input DataInput {
+  fileExtension: String
+  filters: [String!]
+  shapes: [[Int!]!]
+  prefix: String
+  resolution: String
+}
+
+"""
+The input-version of the `Attribute`-type
+"""
+input AttributeInput {
+  type: String
+  layout: String
+  fileExtension: String
+  labels: [String!]
+  shapes: [[Int!]!]
+  prefix: String
+}
+
+"""
+### Manifest
+
+This is the GraphQL representation of the oneseismic `manifest.json`, which is
+stored with each uploaded cube.
+"""
+type Manifest {
+  """
+  Versioning of the manifest itself
+  """
+  formatVersion: Int
+
+  """
+  Original filename of the input SEGY
+  """
+  uploadFilename: String
+
+  """
+  A Globally Unique Identifier for a oneseismic cube
+
+  The guid is used to query the main oneseimic API
+  """
+  guid: String
+
+  """
+  Storage information about the data
+  """
+  data: [Data!]
+
+  """
+  Storage information about additional attributes such as cdpx and cdpy
+  """
+  attributes: [Attribute!]
+
+  """
+  Linenumbers for each dimension of the cube. The labels for each dimension are
+  recorded in `lineLabels`
+  """
+  lineNumbers: [[Int!]!]
+
+  """
+  Labels for each of the linenumber-arrays in `lineNumbers`. Typically
+  'inline', 'crossline' and 'time'
+  """
+  lineLabels: [String!]
+
+  """
+  The minimal sample value in the cube
+  """
+  sampleValueMin: Float
+
+  """
+  The maximal sample value in the cube
+  """
+  sampleValueMax: Float
+  # corners : [Corners!]
```

Comment:
This will be a 1-to-1 mapping of what Sveinung puts into the manifest.

On ingest, we use these values to create a PostGIS polygon which is inserted in the db as a separate column to the manifest. When we do intersection querying from the api, we query against that column, utilizing the PostGIS powers, and not against the manifest itself. 

These fields can be regarded as the underlying data for the geometry stuff, if you need direct access to them instead of doing intersection queries. Again, mostly there for completeness I guess.

But since we havnen't merge the geo section of the manifest yet, I'll remove it for now